navigation/index.tsx

```diff
@@ -103,6 +107,27 @@ function RootNavigator() {
       <Stack.Group screenOptions={{ presentation: "modal" }}>
         <Stack.Screen name="Modal" component={ModalScreen} />
         <Stack.Screen name="NewPost" component={PostScreen} />
+        <Stack.Screen
```

Comment:
as discussed, let's make product details a screen instead. I can take care of it tho bc I have it changed on my end